test/libsolidity/syntaxTests/viewPureChecker/user_operator_with_view_modifier.sol

```diff
@@ -0,0 +1,30 @@
+type Int is uint8;
```

Comment:
This feature has a ton of corner cases. I think we need more tests. Here are some ideas:
- Binding weird things
    - Trying to bind a bound free or library function (e.g. `uint v; using {f} for uint; using {v.f as +} for X`).
    - Trying to bind a function pointer.
    - Trying to bind a builtin function (e.g. `revert()`, `require()`, `blockchash()`, `keccak256()`, `abi.encode()`, `<address>.call()`).
    - Trying to bind an error or an event (both have function types). Or binding an operator to an error/event.
- Trying to call the operator using `x.f()` or `x.+()` syntax after doing `using {f as +} for X`.
- Binding a function with zero arguments.
- Imports
    - Attaching different operators to the same type in different files and then importing the type.
    - Attaching operators to imported types.
    - Operators with `using ... for ... global` (should it affect them?).
- Scoping/shadowing/redefining operators. For example:
    - Attaching a different function as an operator inside a contract when one was already attached at file scope.
    - Redefining the operator as one thing in a base contract and another thing in an inherited one.
    - Binding overloaded functions.
- Trying to redefine other [operators from the table](https://docs.soliditylang.org/en/latest/types.html#order-of-precedence-of-operators) or some weird things like dot, `""`, `()` or `$`.
- Making external calls or using inline assembly within an operator.

Reply:
Binding an operator inside a contract should not be possible, but `using ... global` is something that should be well tested indeed! It should be possible - that's the whole point ;)

Reply:
Ah ok, it is allowed inside a contract. Let's keep it like that then!

Reply:
Great. Most of it is exactly as it should be. Just two things:

> Trying to bind a function pointer.
> I am not sure how to deal with that. Could you put some light on that case?

Something like this:

```solidity
type int is Int;

contract C {
    function(Int, Int) external returns (Int) ptr;
    using {ptr} for Int;
}
```

> Trying to call the operator using `x.f()` or `x.+()` syntax after doing using `{f as +} for X`.

Can you add `x.+()` too? It's weird enough that I guess it could break something.

Reply:
OK, test added for the function pointer.  But, I am not sure about `x.+()`. Does solidity support anything like that?

Reply:
Yeah, we definitely don't want to implement it here. It needs some design discussion anyway.

Fine to just add a proper test for overloading for now (i.e. showing that overloading is not possible).

Reply:
Added test for overloading.